cmd/src/main_test.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestReadConfig(t *testing.T) {
+	makeTempConfig := func(t *testing.T, c config) (string, func()) {
+		data, err := json.Marshal(c)
+		if err != nil {
+			t.Fatal(err)
+		}
+		tmpDir, err := ioutil.TempDir("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		filePath := filepath.Join(tmpDir, "config.json")
+		err = ioutil.WriteFile(filePath, data, 0600)
+		if err != nil {
+			t.Fatal(err)
+		}
+		return filePath, func() { os.RemoveAll(tmpDir) }
```

Comment:
Since you already have `t` at hand, you could do this:

```suggestion
		t.Cleanup(func() { os.RemoveAll(tmpDir) })
		return filePath
```

Reply:
I used `t.Cleanup` originally but `src-cli` is using an older version of Go in the build pipeline.

Let me check how easy it is to update that

Reply:
See comments and commits, I couldn't get Appveyor to use Go 1.14